homeassistant/components/camera/__init__.py

```diff
@@ -411,7 +426,19 @@ class CameraMjpegStream(CameraView):
     url = '/api/camera_proxy_stream/{entity_id}'
     name = 'api:camera:stream'
 
-    @asyncio.coroutine
-    def handle(self, request, camera):
-        """Serve camera image."""
-        yield from camera.handle_async_mjpeg_stream(request)
+    async def handle(self, request, camera):
+        """Serve camera stream, possibly with interval."""
+        interval = request.query.get('interval')
+        if interval is None:
+            await camera.handle_async_mjpeg_stream(request)
+            return
+
+        try:
+            # Compose camera stream from stills
+            interval = float(request.query.get('interval'))
+            await camera.handle_async_still_stream(request, interval)
+            return
+        except (ValueError, HomeAssistantError):
+            return web.Response(status=400)
+
+        return web.Response(status=500)
```

Comment:
This is unreachable, you have a return inside the try and inside the except

Reply:
Wouldn't we reach it when an error is raised other than the one we catch?

Reply:
No, uncaught exceptions blow up the whole function (which is fine, it will trigger a 500)